docs/server-side-search.rst

```diff
@@ -64,3 +64,123 @@ and then click on :guilabel:`Search Analytics`.
    Search analytics demo
 
 .. _Elasticsearch: https://www.elastic.co/products/elasticsearch
+
+API
+---
+
+Search is exposed through our API that's proxied from the domain where your docs are being served.
+This is ``https://docs.readthedocs.io/_/api/v2/search`` for the ``docs`` project, for example.
+
+.. warning::
+
+   This API isn't stable yet, some small things may change in the future.
```

Comment:
I'm leaving this warning since the response will change when we start returning relative paths

Reply:
I wonder if we should do the relative path work in another PR after merging this one, before shipping it? Seems nice to get it all done together if possible.

Reply:
Sure, I can start working in another PR